dash_docs/chapters/advanced_callbacks/index.py

```diff
@@ -89,5 +89,193 @@ def slow_function(input):
+
+    It is important to note that when a Dash app is initially loaded in a web browser by the `dash-renderer` front-end client, its entire callback chain is introspected recursively.
+
+    This allows the `dash-renderer` to predict the order in which callbacks will need to be executed, as callbacks are blocked when their inputs are outputs of other callbacks which have not yet fired. In order to unblock the execution of these callbacks, first callbacks whose inputs are immediately available must be executed. This process helps the `dash-renderer` to minimize the number of network requests it needs to make by making sure it only requests that a callback is executed when all of the callback's inputs have changed.
+
+    Examine the following Dash app:
+    '''),
+
+    rc.Syntax('''
+import dash
+from dash.dependencies import Input, Output
+import dash_html_components as html
+
+app = dash.Dash()
+app.layout = html.Div(
+    [
+        html.Button("execute callback", id="button"),
+        html.Div(children="callback not executed", id="first_output"),
+        html.Div(children="callback not executed", id="second_output"),
+    ]
+)
+
+
+@app.callback(
+    [Output("first_output", "children"), Output("second_output", "children")],
+    [Input("button", "n_clicks")]
+)
+def change_text(n_clicks):
+    if n_clicks is None:
+        return ["n_clicks is None", "n_clicks is None"]
+    return ["n_clicks is " + str(n_clicks), "n_clicks is " + str(n_clicks)]
+
+app.run_server(debug=True)
+
+    '''),
+
+    rc.Markdown('''
+    Notice that when this app is finished being loaded by a web browser and ready for user interaction, the `children` of the `html.Div` components is not the text that is defined when the components are declared in the app's layout but rather is the result of the `change_text()` callback being executed. This is because the "initial call" of the callback occured with `n_clicks` having the value of `None`.
+
+    ### As A Direct Result of User Interaction
+
+    Most frequently, callbacks are executed as a direct result of user interaction, such as clicking a button or selecting an item in a dropdown menu. When such interactions occur, Dash components communicate their new values to the `dash-renderer` front-end client, which then requests that the Dash server execute any callback function that has the newly changed value as input.
+
+    If a Dash app has multiple callbacks, the `dash-renderer` batches requests for callbacks to be executed based on whether or not they can be immediately executed with the newly changed inputs. If several inputs change simultaneously, then requests are made to execute them all.
```

Comment:
```suggestion
    If a Dash app has multiple callbacks, the `dash-renderer` requests callbacks to be executed based on whether or not they can be immediately executed with the newly changed inputs. If several inputs change simultaneously, then requests are made to execute them all.
```

I wouldn't want to give the impression that we merge multiple callbacks into a single request, which "batches requests" might imply. Someday we might do this, but not today :)